src/libLookup/Lookup.cpp

```diff
@@ -1248,6 +1248,243 @@ bool Lookup::ProcessSetSeedPeersFromLookup(const vector<unsigned char>& message,
+
+    uint128_t ipAddr = from.m_ipAddress;
+    Peer requestingNode(ipAddr, portNo);
+    vector<MicroBlock> retMicroBlocks;
+    for (const auto& microBlockId : microBlockIds)
+    {
+
+        const uint64_t& blocknum = microBlockId.first;
+        for (const auto& shard_id : microBlockId.second)
+        {
+            shared_ptr<MicroBlock> mbptr;
+            if (!BlockStorage::GetBlockStorage().GetMicroBlock(blocknum,
+                                                               shard_id, mbptr))
+            {
+                LOG_GENERAL(WARNING,
+                            "Failed to fetch micro block blocknum: "
+                                << blocknum << "Shard ID" << shard_id);
+                continue;
+            }
+            else
+            {
+                retMicroBlocks.push_back(*mbptr);
+            }
+        }
+    }
+
+    vector<unsigned char> retMsg
+        = {MessageType::LOOKUP, LookupInstructionType::SETMICROBLOCKFROMLOOKUP};
+
+    if (retMicroBlocks.size() == 0)
+    {
+        LOG_GENERAL(WARNING, "return size 0 for microblocks");
+        return true;
+    }
+
+    if (!Messenger::SetLookupSetMicroBlockFromLookup(
+            retMsg, MessageOffset::BODY, retMicroBlocks))
+    {
+        LOG_GENERAL(WARNING, "Failed to Process ");
+        return false;
+    }
+
+    P2PComm::GetInstance().SendMessage(requestingNode, retMsg);
+    return true;
+}
+
+bool Lookup::ProcessSetMicroBlockFromLookup(
+    const vector<unsigned char>& message, unsigned int offset,
+    [[gnu::unused]] const Peer& from)
+{
+    //[numberOfMicroBlocks][microblock1][microblock2]...
+
+    vector<MicroBlock> mbs;
+    mbs.clear();
+    if (!Messenger::GetLookupSetMicroBlockFromLookup(message, offset, mbs))
+    {
+        LOG_GENERAL(WARNING, "Failed to process");
+    }
+
+    for (const auto& mb : mbs)
+    {
+        LOG_GENERAL(INFO,
+                    "[SendMB]"
+                        << " Recvd " << mb.GetHeader().GetBlockNum()
+                        << " shard:" << mb.GetHeader().GetShardID());
+    } //do something with mb
+
+    return true;
+}
+
+void Lookup::SendGetMicroBlockFromLookup(
+    const map<uint64_t, vector<uint32_t>>& mbInfos)
+{
+    vector<unsigned char> msg
+        = {MessageType::LOOKUP, LookupInstructionType::GETMICROBLOCKFROMLOOKUP};
+
+    if (mbInfos.size() == 0)
+    {
+        LOG_GENERAL(INFO, "No microBlock requested");
+        return;
+    }
+
+    if (!Messenger::SetLookupGetMicroBlockFromLookup(
+            msg, MessageOffset::BODY, mbInfos,
+            m_mediator.m_selfPeer.m_listenPortHost))
+    {
+        LOG_GENERAL(WARNING, "Failed to process");
+        return;
+    }
+
+    SendMessageToRandomLookupNode(msg);
+}
+
+void Lookup::CommitMicroBlockStorage()
+{
+    lock_guard<mutex> g(m_mutexMicroBlocksBuffer);
+    const uint64_t& currentEpoch = m_mediator.m_currentEpochNum;
+
+    for (uint64_t i = 1; i <= currentEpoch; i++)
```

Comment:
This loop will be very very long if have large epoch num in the future. you can just use an iterate to loop among the map for only existing value

Reply:
noted